### DISCUSSION TITLE
UUID from name

### DISCUSSION THREAD

**Discussion by LazyGamer1111** [Minestom/Minestom#1371] [Help]
How can i get a UUID from a name?
i want this to make a enity with entity type player.
I dont have any code because i dont know how i could start with it.

**Comment by GreatWyrm** [comment]
If you just want to make a Player entity that isn't controlled by a user, I would recommend [FakePlayer](https://javadoc.minestom.net/net/minestom/server/entity/fakeplayer/FakePlayer.html#initPlayer(java.util.UUID,java.lang.String,net.minestom.server.entity.fakeplayer.FakePlayerOption,java.util.function.Consumer)).

If you're looking to get player data based on a username, you will need [MojangUtils.fromUsername](https://javadoc.minestom.net/net/minestom/server/utils/mojang/MojangUtils.html#fromUsername(java.lang.String)).

**Reply by LazyGamer1111** [reply]
a fake player can accept packets, i just want a clickable npc, so an npc like i said is better.

**Reply by GreatWyrm** [reply]
I mean, you didn't say earlier that an npc is better, but okay.
FakePlayer can still be used as an NPC, and it may be easier on you.

But you could create your own fake player class, with something like:

```java
public class YourFakePlayer extends LivingEntity {

    public YourFakePlayer() {
         super(EntityType.PLAYER, UUID.randomUUID()); 
    }
}
```

You will have to do some packet manipulation because players are special entities (you need to send the addPlayer/removePlayer packets upon creating/removing the entity), and probably some other things that I don't remember off the top of my head.
